src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_test.go

```diff
@@ -115,3 +116,96 @@ func TestDetermineFilepath(t *testing.T) {
 		})
 	}
 }
+
+func TestParseJournalTime(t *testing.T) {
+	for _, test := range []struct {
+		name   string
+		input  string
+		exp    time.Time
+		expErr bool
+		bound  bool
+	}{
+		{
+			name:  "YYYY-MM-DD",
+			input: "2023-11-30",
+			exp:   time.Date(2023, 11, 30, 0, 0, 0, 0, time.Local),
+		}, {
+			name:  "YYYY-MM-DD with leading and trailing space",
+			input: " 2023-11-30 ",
+			exp:   time.Date(2023, 11, 30, 0, 0, 0, 0, time.Local),
+		}, {
+			name:  "YYYY-MM-DD HH:MM:SS",
+			input: "2023-11-30 12:13:14",
+			exp:   time.Date(2023, 11, 30, 12, 13, 14, 0, time.Local),
+		}, {
+			name:  "No seconds: YYYY-MM-DD HH:MM",
+			input: "2023-11-30 12:13",
+			exp:   time.Date(2023, 11, 30, 12, 13, 0, 0, time.Local),
+		}, {
+			name:   "YYYY-MM-DD HH No Minutes",
+			input:  "2023-11-30 12",
+			expErr: true, // This is not specified in man journalctl, so we err.
+		}, {
+			name:   "YYYY-MM-DD HH:MM:S With 1 second",
+			input:  "2023-11-30 12:13:1",
+			expErr: true, // This is not specified in man journalctl, so we err.
+		}, {
+			name:   "YYYY-MM-DD HH:MM: with extra :",
+			input:  "2023-11-30 12:13:",
+			expErr: true, // This is an error in journalctl too.
+		}, {
+			name:  "now",
+			input: "now",
+			bound: true,
+			exp:   time.Now(),
+		}, {
+			name:  "today",
+			input: "today",
+			bound: true,
+			exp:   time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.Local),
+		}, {
+			name:  "yesterday",
+			input: "yesterday",
+			bound: true,
+			exp:   time.Date(time.Now().Year(), time.Now().Month(), time.Now().AddDate(0, 0, -1).Day(), 0, 0, 0, 0, time.Local),
+		}, {
+			name:   "unrecognized text",
+			input:  "todayzz",
+			expErr: true,
+		}, {
+			name:  "+ relative time",
+			input: "2h",
+			bound: true,
+			exp:   time.Now().Add(2 * time.Hour),
+		}, {
+			name:  "- relative time",
+			input: "-5day",
+			bound: true,
+			exp:   time.Now().AddDate(0, 0, -5),
+		}, {
+			name:   "unrecognized relative time",
+			input:  "-5trillions",
+			expErr: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			parsedTime, err := parseJournalTime(test.input)
+			if test.expErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			// Used when we need to test input susceptible to time drift between
+			// test declaration and test execution (i.e: relative times)
+			if test.bound {
+				if parsedTime.Before(test.exp) && test.exp.Sub(parsedTime) > time.Minute ||
+					test.exp.Before(parsedTime) && parsedTime.Sub(test.exp) > time.Minute {
+					t.Errorf("error: got %v not within 1min of expected %v", parsedTime.String(), test.exp.String())
+				}
+				return
+			}
+			require.Equal(t, test.exp, parsedTime)
```

Comment:
One way to get around this bounded logic is to require `time.Time` as an arg to `parseJournalTime`, and then in tests, pass a defined time.